build/images/Dockerfile.build.windows

```diff
@@ -1,30 +1,10 @@
 ARG GO_VERSION
-FROM mcr.microsoft.com/powershell:lts-nanoserver-1809 as cni-binaries-windows
-SHELL ["pwsh", "-NoLogo", "-Command", "$ErrorActionPreference = 'Stop'; $ProgressPreference = 'SilentlyContinue';"]
-
-ENV CNI_PLUGINS="./host-local.exe"
-
-WORKDIR /opt/cni/bin
-
-RUN curl.exe -LO https://github.com/containernetworking/plugins/releases/download/v0.8.1/cni-plugins-windows-amd64-v0.8.1.tgz; \
-    tar -xzf cni-plugins-windows-amd64-v0.8.1.tgz  -C C:\opt\cni\bin ${env:CNI_PLUGINS}; \
-    rm cni-plugins-windows-amd64-v0.8.1.tgz
 
+FROM golang:${GO_VERSION}-nanoserver as antrea-golang
 
-FROM golang:${GO_VERSION} as antrea-build-windows
+FROM gavinxin/base-windows:20230328 as antrea-build-windows
 
-WORKDIR /
-
-# Install 7zip, git-for-windows, mingw64 to support "make tool"
-RUN curl.exe -LO https://www.7-zip.org/a/7z2107-x64.exe; \
-    cmd /c start /wait 7z2107-x64.exe /S; \
-	del 7z2107-x64.exe; $env:Path=$env:Path + ';C:/Program Files/7-Zip'; [Environment]::SetEnvironmentVariable('PATH', $env:Path, [EnvironmentVariableTarget]::Machine)
-RUN curl.exe -LO https://cfhcable.dl.sourceforge.net/project/mingw-w64/Toolchains%20targetting%20Win64/Personal%20Builds/mingw-builds/8.1.0/threads-posix/seh/x86_64-8.1.0-release-posix-seh-rt_v6-rev0.7z; \
-    7z x x86_64-8.1.0-release-posix-seh-rt_v6-rev0.7z; $env:Path = 'C:\mingw64\bin;' + $env:Path; cp C:\mingw64\bin\mingw32-make.exe  C:\mingw64\bin\make.exe; \
-	del x86_64-8.1.0-release-posix-seh-rt_v6-rev0.7z; [Environment]::SetEnvironmentVariable('PATH', $env:Path, [EnvironmentVariableTarget]::Machine)
-RUN curl.exe -LO https://github.com/git-for-windows/git/releases/download/v2.26.2.windows.1/Git-2.26.2-64-bit.exe; \
-    cmd /c start /wait Git-2.26.2-64-bit.exe /VERYSILENT /NORESTART /NOCANCEL /SP- /NOICONS /COMPONENTS='icons,icons\quicklaunch,ext,ext\reg,ext\reg\shellhere,ext\reg\guihere,assoc,assoc_sh' /LOG; \
-	$env:Path='C:\Program Files\Git\cmd;C:\Program Files\Git\usr\bin;' + $env:Path; [Environment]::SetEnvironmentVariable('PATH', $env:Path, [EnvironmentVariableTarget]::Machine)
+COPY --from=antrea-golang ["C:\\\\Program Files\\\\Go", "C:\\\\Program Files\\\\Go"]
```

Comment:
maybe you should just install the Go binary in the `base-windows` image?

Reply:
Actually, I saw that base-ubuntu for Ubuntu also has OVS version dependency, and it has different image tags according to OVS changes. So it is acceptable that base-windows image has dependency for Go versions, right @antoninbas ?

Reply:
One thing we need to consider is automatically updating the base image when the Windows image (`mcr.microsoft.com/windows/servercore:1809`) is updated, e.g. with security updates. This is what we do on Linux.

Thinking about it more, I don't think we should rely on manual updates for the `base-windows` image. It should all be handled automatically in CI.

If you look at the toolchain for Linux / Ubuntu, you will see that we always "rebuild" the image, but in practice the build is really fast because we can benefit from caching when there are no changes to the base image or to the version of dependencies. Going this route, we can also include Go in the base image.

The following approach would make sense to me:
- generate a tag `<TAG>` deterministically based on the version of dependencies: go version, nanoserver version, CNI binaries version. Something like this would work to generate `<TAG>`: `cat go-version cni-binaries-version nanoserver-version | md5sum | awk '{print $1}'`.
- build and push the image automatically in CI with the correct tag. We need to make sure that caching works properly when dependencies do not change.
- build the `antrea-windows` image using the correct `base-windows:<TAG>`

Reply:
@antoninbas  Thanks for the suggestions. I think there is still one problem here. Our windows testbeds will revert snapshot at each CI test for restoring environment. So I think we can't use docker cache to speed up image building in this case. My original plan was to build a image that requires little update and save its building cache into windows snapshot when creating testbed. Then we can reuse cache of base environment and build other components which may have dependency for other version parameters each time. The initial reason for this PR is to speed up windows building. If we have to rebuild base image every time I think it may not be different from the previous solution, or even slower.

Reply:
> I think there is still one problem here. Our windows testbeds will revert snapshot at each CI test for restoring environment.

The Linux build pulls the image from the registry first and then uses it as a cache, thanks to the `--cache-from` command line option: https://github.com/antrea-io/antrea/blob/main/build/images/base/build.sh. You can check out some recent builds in CI to see that the image is not rebuilt from scratch every time.

Reply:
If Windows docker builds don't work the same way, let me know